mission_control/navigator_missions/nav_missions/start_gate.py

```diff
@@ -1,66 +1,277 @@
+    #req.name = 'r_gate'
+    #right = yield serv(req)
+    #buoys = [Buoy.from_srv(left), Buoy.from_srv(right)]
+    print buoys
+    # Display points of the buoys]
+    points = [navigator_tools.numpy_to_point(b.position) for b in buoys]
+
+    pose = yield navigator.tx_pose
+    buoys = sorted(buoys, key=lambda buoy: buoy.distance(pose[0]))
+
+    # print pose
+    # print buoys
+
+    print "\nSTART_GATE: Be advised... Doing checks"
+
+    # See if the closest two buoys are the right colors ============================================
+    if not (buoys[0].color in ['green', 'red'] and buoys[1].color in ['green', 'red']):
+        print "START_GATE: Error 1 - Not the right colors. ({},{})".format(buoy[0].color,
+                                                                           buoy[1].color)
+        result.success = False
+        #return_with(result)
+
+    # Okay, now we need to see if the buoys are approx 10m apart ===================================
+    nominal_dist = 10  # m
+    variance = 5  #m
+    dist_btwn_buoys = np.linalg.norm(buoys[0].position[:2] - buoys[1].position[:2])
+    if not (nominal_dist - variance < dist_btwn_buoys < nominal_dist + variance):
+        print "START_GATE: Error 2 - Not the right distance apart. {}".format(dist_btwn_buoys)
+        #result.success = False
+        #return_with(result)
+
+    # Made it this far, make sure the red one is on the left and green on the right ================
+    #t = yield navigator.tf_listener.get_transform("/base_link", "/enu", navigator.nh.get_time())
+    class t(object):
+        # Temp
+        @classmethod
+        def transform_point(self, point):
+            return point
+    bl_buoys = [t.transform_point(buoy.position) for buoy in buoys[:2]]
+
+    # Angles are w.r.t positive x-axis. Positive is CCW around the z-axis.
+    angle_buoys = np.array([np.arctan2(buoy[1], buoy[0]) for buoy in bl_buoys])
+    left, right = buoys[np.argmax(angle_buoys)], buoys[np.argmin(angle_buoys)]
+    if left.color == "green" and right.color == "red":
+        print "START_GATE: Error 3 - Not the right colors. (l-{}, r-{})".format(left.color,
+                                                                                right.color)
+        #result.success = False
+        #return_with(result)
+
+    # Lets get a path to go to
+    between_vector, direction_vector = get_path(left, right)
+    mid_point = left.position + between_vector / 2
+
+    #print mid_point
+    setup = mid_point - direction_vector * 20
+    target = setup + direction_vector * 90
+
+    ogrid = OgridFactory(left.position, right.position, pose[0], target)
+    msg = ogrid.get_message()
+
+    # Put the target into the point cloud as well
+    points.append(navigator_tools.numpy_to_point(target))
+    pc = PointCloud(header=navigator_tools.make_header(frame='/enu'),
+                    points=np.array(points))
+    for i in range(50):
+        yield navigator._point_cloud_pub.publish(pc)
```

Comment:
Its used to visualize different points used in the start gate mission. Its published 50 times because occasionally by only publishing once, it will not get received by rviz.